es5-sham.js

```diff
@@ -65,7 +65,7 @@ if (!Object.getPrototypeOf) {
         /* eslint-enable no-proto */
         if (proto || proto === null) {
             return proto;
-        } else if (object.constructor) {
+        } else if (typeof object.constructor === 'function') {
```

Comment:
Will this not return a false positive with regexes in some older engines?

I think we need an `Object#toString` check here instead.